go/adbc/driver/databricks/statement.go

```diff
@@ -0,0 +1,424 @@
+
+	// req is the request to be sent to the Databricks SQL API
+	req *sql.ExecuteStatementRequest
+
+	// statementId is the ID of the current statement in the Databricks SQL API
+	statementId string
+	// The manifest from the first StatementResponse
+	manifest *sql.ResultManifest
+}
+
+func NewStatement(conn *connectionImpl) (adbc.Statement, error) {
+	return &statement{
+		alloc: conn.Alloc,
+		conn:  conn,
+
+		req: &sql.ExecuteStatementRequest{
+			// Configurable via OptionIntByteLimit
+			ByteLimit: 0,
+			Catalog:   conn.catalog,
+			// Arrow responses can't be INLINE, so we always use EXTERNAL_LINKS
+			Disposition: sql.DispositionExternalLinks,
+			Format:      sql.FormatArrowStream,
+			// Continue execution asynchronously after the short wait timeout
+			OnWaitTimeout: sql.ExecuteStatementRequestOnWaitTimeoutContinue,
+			// TODO: set this when binding is implemented
+			Parameters: []sql.StatementParameterListItem{},
+			// Configurable via OptionIntRowLimit
+			RowLimit: 0,
+			Schema:   conn.dbSchema,
+			// Populated by SetSqlQuery()
+			Statement: "",
+			// TODO: make WaitTimeout configurable
+			// Short timeout to get to the first record batch or just
+			// the statement ID for polling later. Must be in [5s-50s].
+			WaitTimeout: "5s",
+			WarehouseId: conn.client.Config.WarehouseID,
+		},
+
+		statementId: "",
+		manifest:    nil,
+	}, nil
+}
+
+func (stmt *statement) resetStatement(sql string) error {
+	// TODO: think about cancelation and stmt re-use
+	stmt.req.Parameters = nil
+	stmt.req.Statement = sql
+	stmt.statementId = ""
+	stmt.manifest = nil
+	return nil
+}
+
+func (stmt *statement) SetOption(key, val string) error {
+	switch key {
+	case OptionBoolTruncated:
+		return adbc.Error{
+			Code: adbc.StatusInvalidArgument,
+			Msg:  fmt.Sprintf("[Databricks] `%s` property is read-only", key),
+		}
+	default:
+		return adbc.Error{
+			Code: adbc.StatusInvalidArgument,
+			Msg:  fmt.Sprintf("[Databricks] Unknown statement string type option `%s`", key),
+		}
+	}
+}
+
+func (stmt *statement) GetOption(key string) (string, error) {
+	switch key {
+	case OptionBoolTruncated:
+		truncated := false
+		if stmt.manifest != nil {
+			truncated = stmt.manifest.Truncated
+		}
+		return strconv.FormatBool(truncated), nil
+	default:
+		val, err := stmt.conn.GetOption(key)
+		if err == nil {
+			return val, nil
+		}
+		return "", err
+	}
+}
+
+func (stmt *statement) SetOptionInt(key string, value int64) error {
+	switch key {
+	case OptionIntRowLimit:
+		stmt.req.RowLimit = value
+	default:
+		return adbc.Error{
+			Code: adbc.StatusInvalidArgument,
+			Msg:  fmt.Sprintf("[Databricks] Unknown statement string type option `%s`", key),
+		}
+	}
+	return nil
+}
+
+func (stmt *statement) GetOptionInt(key string) (int64, error) {
+	switch key {
+	case OptionIntRowLimit:
+		if stmt.req != nil {
+			return stmt.req.RowLimit, nil
+		}
+		return 0, nil
+	default:
+		val, err := stmt.conn.GetOptionInt(key)
+		if err == nil {
```

Comment:
can this just return `val, err`? if 0 for some spec that it has to be with err

Reply:
oh my, 4 lines here can be come a single return